sky/serve/core.py

```diff
@@ -262,6 +263,104 @@ def up(
             f'short time.{style.RESET_ALL}')
 
 
+@usage_lib.entrypoint
+def update(task: 'sky.Task',
+           service_name: str,
+           mixed_replica_versions: bool = False) -> None:
+
+    cluster_status, handle = backend_utils.is_controller_up(
+        controller_type=controller_utils.Controllers.SKY_SERVE_CONTROLLER,
+        stopped_message='All services should have terminated.')
+
+    if handle is None or handle.head_ip is None:
+        # The error message is already printed in
+        # backend_utils.is_controller_up
+        # TODO(zhwu): Move the error message into the exception.
+        with ux_utils.print_exception_no_traceback():
+            raise exceptions.ClusterNotUpError(message='',
+                                               cluster_status=cluster_status)
+
+    backend = backend_utils.get_backend_from_handle(handle)
+    assert isinstance(backend, backends.CloudVmRayBackend)
+
+    code = serve_utils.ServeCodeGen.get_service_status([service_name])
+
+    try:
+        returncode, serve_status_payload, _ = backend.run_on_head(
+            handle, code, require_outputs=True, stream_logs=False)
+    except exceptions.CommandError as e:
+        raise RuntimeError(e.error_msg) from e
+
+    service_statuses = serve_utils.load_service_status(serve_status_payload)
+    if len(service_statuses) == 0:
+        with ux_utils.print_exception_no_traceback():
+            raise RuntimeError(f'Cannot find service {service_name!r}.')
+
+    assert len(service_statuses) == 1
+    service_record = service_statuses[0]
+    prompt = None
+    if service_record is None:
+        prompt = (f'Service {service_name!r} not exist. '
+                  'To create a new service, use `sky serve up`.')
+    elif (service_record['status'] ==
+          serve_state.ServiceStatus.CONTROLLER_FAILED):
+        prompt = (f'Service {service_name!r} has a failed controller. '
+                  'Please clean up the service and try again.')
+    elif (service_record['status'] == serve_state.ServiceStatus.CONTROLLER_INIT
+         ):
+        prompt = (f'Service {service_name!r} is still initializing '
+                  'its controller. Please try again later.')
+    if prompt is not None:
+        with ux_utils.print_exception_no_traceback():
+            raise RuntimeError(prompt)
```

Comment:
Could we move those checks on the controller? Reason: to reduce #SSH connections fro 2 to 1.

Reply:
you need the ssh connection here because you need to get 
```
service_statuses = serve_utils.load_service_status(serve_status_payload)
```
for version and controller port.

Reply:
We don't need those variables locally right? We could just get those in the controller and move all logic into controller. Here we only need to sync a new version of yaml and passed a service name to the controller; and we figure out all other info on the controller

Reply:
I still think keeping them separate is better. It keeps them functionally separate (e.g. handling returned errors). 
(1) check service status (2) send to serve controller for updating the status.
Each requires a ssh, but ssh does not have that much overhead. 
we can mention this in our discussion as well

Reply:
Based on offline discussion with @Michaelvll , let's keep them separate